packages/connectivity/android/src/main/java/io/flutter/plugins/connectivity/ConnectivityPlugin.java

```diff
@@ -62,7 +66,31 @@ public void onCancel(Object arguments) {
     receiver = null;
   }
 
-  private static String getNetworkType(int type) {
+  private NetworkInfo getNetworkInfo(Context context){
+    ConnectivityManager cm =
+            (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+    return cm.getActiveNetworkInfo();
+  }
+
+  @SuppressWarnings("deprecated")
+  private String getNetworkType(int type) {
+    if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+      Network network = manager.getActiveNetwork();
+      NetworkCapabilities capabilities = manager.getNetworkCapabilities(network);
+      if (capabilities != null) {
```

Comment:
nits:
it'S preferred to return early e.g.:
```dart
if (capabilities == null) {
  return "none";
}
if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_WIFI)
            || capabilities.hasTransport(NetworkCapabilities.TRANSPORT_ETHERNET)) {
          return "wifi";
}
if (capabilities.hasTransport(NetworkCapabilities.TRANSPORT_CELLULAR)) {
   return "mobile";
}
```

Reply:
Okay, I have done and pushed new changes

Reply:
Can you help me with test format? I can't run the flutter tool format on my pc. I am getting this [issue ](https://github.com/flutter/flutter/issues/15381)